### PR TITLE
Strip personally identifiable information from user table for unautho…

### PR DESCRIPTION
**Pull request by acinader**
…rized users.

- add a config option to explicitly enumerate pii fields beyond email
- in query controller, strip pii of user table results before sending out the door.

fixes: #3155

### CODE REVIEW

**Comment by acinader** [comment]
oops. forgot to commit the unit test...now added.

File: src/RestQuery.js [hunk 1]
```diff
@@ -407,7 +417,7 @@ RestQuery.prototype.runFind = function(options = {}) {
       if (this.className === '_User') {
         for (var result of results) {
           delete result.password;
-
+          cleanResultOfSensitiveUserInfo(result, this.auth, this.config);
```
Comment (flovilmart) [inline src/RestQuery.js]: seems that authData is pretty much the same :)

File: src/RestQuery.js [hunk 2]
```diff
@@ -386,6 +386,16 @@ RestQuery.prototype.replaceDontSelect = function() {
   })
 };
 
+const cleanResultOfSensitiveUserInfo = function (result, auth, config) {
+  if (auth.isMaster || ( auth.user && auth.user.id === result.objectId)) {
```
Comment (flovilmart) [inline src/RestQuery.js]: Small nit: ```  || ( auth.user``` 
-----------------^  remove space

**Comment by flovilmart** [comment]
@acinader just a few small things that you could address. Otherwise, LGTM!

**Comment by acinader** [comment]
@flovilmart 

I added a second commit with what I think you have in mind for fixes the authdata?  I like having the logic pulled out of the loop into their own functions.  I also like having them be two, small function instead of combining.

Also, if this looks good, let me know and I'll squash the commits.  The squash and merge button is the devil.

**Comment by flovilmart** [comment]
The squash and merge button is fine :)

**Comment by acinader** [comment]
@flovilmart

1. squashed
2. au contraire the squash and merge wreaks havoc on my local workflow. changes all the hashes so 3. i can't easily tell which branches have been merged (need to google some more.)
looking into why the build is busted (it wasn't me!) ;)

**Comment by flovilmart** [comment]
2. Right, it rewrites the history, but that's nice. As it also adds on the top of master.

Now that you're a contributor, you don't need your fork for your commits you can push branches directly

**Comment by acinader** [comment]
noted on the fork stuff.  any reason for me to resubmit this not from my fork?

**Comment by flovilmart** [comment]
No reason!

**Comment by virtualtoy** [comment]
Is there an easy way to include email to user queries? `userSensitiveFields` is concatenated internally with default `['email']`. I'd expect that parse server respects `userSensitiveFields` if it's specified

**Pull request merged**